modules/photo/src/layerModelBlending.cpp

```diff
@@ -0,0 +1,201 @@
+//  copy or use the software.
+//
+//
+//                           License Agreement
+//                For Open Source Computer Vision Library
+//
+// Copyright (C) 2013, OpenCV Foundation, all rights reserved.
+// Third party copyrights are property of their respective owners.
+//
+// Redistribution and use in source and binary forms, with or without modification,
+// are permitted provided that the following conditions are met:
+//
+//   * Redistribution's of source code must retain the above copyright notice,
+//     this list of conditions and the following disclaimer.
+//
+//   * Redistribution's in binary form must reproduce the above copyright notice,
+//     this list of conditions and the following disclaimer in the documentation
+//     and/or other materials provided with the distribution.
+//
+//   * The name of the copyright holders may not be used to endorse or promote products
+//     derived from this software without specific prior written permission.
+//
+// This software is provided by the copyright holders and contributors "as is" and
+// any express or implied warranties, including, but not limited to, the implied
+// warranties of merchantability and fitness for a particular purpose are disclaimed.
+// In no event shall the Intel Corporation or contributors be liable for any direct,
+// indirect, incidental, special, exemplary, or consequential damages
+// (including, but not limited to, procurement of substitute goods or services;
+// loss of use, data, or profits; or business interruption) however caused
+// and on any theory of liability, whether in contract, strict liability,
+// or tort (including negligence or otherwise) arising in any way out of
+// the use of this software, even if advised of the possibility of such damage.
+//
+//M*/
+
+#include "precomp.hpp"
+#include "opencv2/photo.hpp"
+
+using namespace std;
+using namespace cv;
+
+#define EPSILON      1e-6f
+
+#define SAFE_DIV_MIN EPSILON
+#define SAFE_DIV_MAX (1.0f / SAFE_DIV_MIN)
+
+#define	CLAMP(f,min,max)	((f)<(min)?(min):(f)>(max)?(max):(f))
+
+namespace cv
+{
+/*  local function prototypes  */
+static inline float	safe_div(float a, float b);
+/* returns a / b, clamped to [-SAFE_DIV_MAX, SAFE_DIV_MAX].
+ * if -SAFE_DIV_MIN <= a <= SAFE_DIV_MIN, returns 0.
+ */
+static inline float safe_div(float a, float b)
+{
+    float result = 0.0f;
+    if (fabsf(a) > SAFE_DIV_MIN)
+    {
+        result = a / b;
+        result = CLAMP(result, -SAFE_DIV_MAX, SAFE_DIV_MAX);
+    }
+    return result;
+}
+CV_EXPORTS_W void layerModelBlending(InputArray _target, InputArray _blend, OutputArray _dst, int flag)
+{
+    Mat target = _target.getMat();
+    Mat blend = _blend.getMat();
+    Mat dst = _dst.getMat();
+
+    for (int index_row = 0; index_row < target.rows; index_row++)
+        for (int index_col = 0; index_col < target.cols; index_col++)
+            for (int index_c = 0; index_c < 3; index_c++)
+                switch (flag)
```

Comment:
I believe per-pixel loop with a switch on every iteration is not efficient way of doing it. Is it possible to use existing operations to implement blending, e.g. [addWeighted](https://docs.opencv.org/master/d2/de8/group__core__array.html#gafafb2513349db3bcff51f54ee5592a19)?

Reply:
“addWeighted” is very worth learning, and I also know there are some methods to improve the speed loop。However that will make the code more difficult to read, and Less intuitive。 I am not sure if it is scientific to do so?@mshabunin

Reply:
We try to focus on efficiency, if you want to demonstrate library usage it is better to create a sample application or tutorial. There is tutorial about pixel-wise operations: https://docs.opencv.org/master/db/da5/tutorial_how_to_scan_images.html and one about blending: https://docs.opencv.org/master/d5/dc4/tutorial_adding_images.html . C++ samples are here: https://github.com/opencv/opencv/tree/master/samples/cpp

Reply:
> tutorial
The PR has been updated.
I can barely write sample code that meets the quality requirements, but the tutorial is relatively rusty, especially my existing editor does not recognize the markdown format on github very well, I hope to give some knowledge. I very much hope that my code can be integrated into OpenCV, and through this process to grow my relevant experience.3Q

Reply:
A supporting demo has been uploaded, and the relevant parts of the comments have also been modified. Thanks for suggestions!